weldx/core.py

```diff
@@ -540,3 +543,41 @@ def units(self) -> str:
         if isinstance(self._data, xr.DataArray):
             return self._data.data.units
         return self._units
+
+
+# ExternalFile -------------------------------------------------------------------------
+
+
+class ExternalFileBuffer:
+    """Handles external files."""
+
+    def __init__(self, path, file_system=None, buffer: np.ndarray = None):
+        if isinstance(path, str):
+            path = Path(path)
+
+        self._filename = path.name
+
+        if buffer is None:
+            if file_system is None:
+                with OSFS(path.parent.absolute().as_posix()) as file_system:
+                    self._buffer = np.frombuffer(
+                        file_system.readbytes(self._filename), dtype=np.int8
```

Comment:
not sure if this changes anything but maybe `np.uint8` is safer here

Reply:
You're probably right. Missed that when I copied the line from Stackoverflow.